core/src/banking_stage.rs

```diff
@@ -510,14 +524,36 @@ impl BankingStage {
         let num_to_commit = num_to_commit.unwrap();
 
         if num_to_commit != 0 {
-            bank.commit_transactions(
-                txs,
-                None,
-                &mut loaded_accounts,
-                &results,
-                tx_count,
-                signature_count,
-            );
+            let transaction_statuses = bank
+                .commit_transactions(
+                    txs,
+                    None,
+                    &mut loaded_accounts,
+                    &results,
+                    tx_count,
+                    signature_count,
+                )
+                .1;
+            if let Some(blocktree) = maybe_blocktree {
```

Comment:
What kind of overhead does `blocktree.write_transaction_status()` have?   This will block banking stage so if it's non-trivial then perhaps we need a channel to offload the writes to a separate thread.

Reply:
Yeah, that approach would also save some of the duplication between replay_stage and banking_stage. I'll rework.